### ISSUE TITLE
idea: update program feature

### ISSUE THREAD

**Issue by edmundlaugasson** [aardappel/treesheets#48]
There is an idea to update program e.g. from Help->Check updates menu. It would then download new version and replace current installation. Also configurable to automatically update and periodically (also period can be chosen e.g. in days) check updates. E.g. like [Telegram](https://telegram.org/) does. There could be configurable setting whether updating would be automatic or manual.

**Comment by aardappel** [comment]
That be convenient, yes, but a lot of work to get right for a multi-platform program.

**Comment by edmundlaugasson** [comment]
I agree that there might be a lot of work. Just for example - I noticed that Telegram has only two binaries - one is program and another is updater. So the policy is the same on different platforms - just download the main binary and replace the current one. But it works quite smoothly - in Telegram interface there will appear "Update" and arrow in top right corner so the user will know that if to click to that corner, the program will be updated and reopened automatically. As Telegram is available [also here](https://github.com/telegramdesktop/tdesktop) then probably there would be possible to copy that model.

**Comment by aardappel** [comment]
Note that Telegram is a funded company. TreeSheets is a hobby project of a guy with a day job :)

**Comment by edmundlaugasson** [comment]
Therefore I proposed an idea to possibly copy their components as it is also freely available in GitHub.

**Comment by aardappel** [comment]
Just because code is available on github doesn't mean that using it in a different code base is going to be easy :)

**Comment by bvazx** [comment]
Since 2015, so much time has passed... And yet, this especially important feature still hasn't been added... I think, given the fairly frequent releases of TreeSheets, it really needs to be implemented!

**Comment by aardappel** [comment]
@bvazx this is an open source project supported by volunteers, not a commercial application, so there is no such thing as "it needs to implemented".
